### PR TITLE
Event put and post #58

### LINKED ISSUE #58: Write POST route for events
For creating a new game.

### PR DESCRIPTION
**Pull request by brenmeyer2245**
Close #58, #60 

Created Post and Put routes with tests
-Created Search By Name Class Method on Event Model

### CODE REVIEW

**Comment by pckilgore** [comment]
There seems to be a lot of code in here that is already on master?  What files should we be looking at?

File: server/api/events.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+const router = require('express').Router()
+const {Event} = require('../db/models')
+module.exports = router
+
+router.get('/', async (req, res, next) => {
```
Comment (pckilgore) [inline server/api/events.js]: Ahh, the missing GET routes!

File: server/api/events.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+router.get('/:eventId', async (req, res, next) => {
+  try {
+    const events = await Event.findById(req.params.eventId)
+    res.send(events)
+  } catch (err) {
+    next(err)
+  }
+})
+
+router.get('/name/:eventName', async (req, res, next) => {
```
Comment (pckilgore) [inline server/api/events.js]: This is cool! I didn't know you could do this! What happens when the name has spaces or weird characters though?

File: server/api/events.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+//This route takes in the entire object and changes the entire object
+//Presumes that values meant to remain consistent will be passed in
+router.put('/:eventId', async (req, res, next) => {
+  const {name, status, description, date, location} = req.body
+
+  const targetEvent = await Event.findById(req.params.eventId)
+  //if values are not passed in through req.body, set values to null
+  await targetEvent.update({
+    name: name || null,
```
Comment (pckilgore) [inline server/api/events.js]: We need error handing here right?  For example, if name is null we'll get an error we need to pass to next.

File: server/api/events.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    status: status || null,
+    description: description || null,
+    date: date || null,
+    location: location || null
+  })
+  res.send(targetEvent)
+})
+
+//This route Specifically changes status
+router.put('/:eventId/status', async (req, res, next) => {
```
Comment (pckilgore) [inline server/api/events.js]: Same, don't we need some error handling in case a bad status is sent? (Not in the enum)

File: server/api/events.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  const {status} = req.body
+  const targetEvent = await Event.findById(req.params.eventId)
+  await targetEvent.update({status})
+  res.send(targetEvent)
+})
+
+//Specifically changes the date for event
+router.put('/:eventId/date', async (req, res, next) => {
+  const {date} = req.body
```
Comment (pckilgore) [inline server/api/events.js]: Same, I'd add some error catching/handling.

File: server/api/events.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  const targetEvent = await Event.findById(req.params.eventId)
+  await targetEvent.update({date})
+  res.send(targetEvent)
```
Comment (pckilgore) [inline server/api/events.js]: If we are sending data back, don't we need to set the {returning:true} option?

File: server/api/events.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  })
+  res.send(targetEvent)
+})
+
+//This route Specifically changes status
+router.put('/:eventId/status', async (req, res, next) => {
+  const {status} = req.body
+  const targetEvent = await Event.findById(req.params.eventId)
+  await targetEvent.update({status})
+  res.send(targetEvent)
```
Comment (pckilgore) [inline server/api/events.js]: If we are sending data back, don't we need to set the {returning:true} option?

File: server/api/events.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+  const targetEvent = await Event.findById(req.params.eventId)
+  //if values are not passed in through req.body, set values to null
+  await targetEvent.update({
+    name: name || null,
+    status: status || null,
+    description: description || null,
+    date: date || null,
+    location: location || null
+  })
+  res.send(targetEvent)
```
Comment (pckilgore) [inline server/api/events.js]: If we are sending data back, don't we need to set the {returning:true} option?

File: server/api/events.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+router.post('/', async (req, res, next) => {
+  const {name, status, description, date, location} = req.body
+  const newEvent = await Event.create({
+    name,
+    status,
+    description,
+    date,
+    location
+  })
+  res.send(newEvent)
```
Comment (pckilgore) [inline server/api/events.js]: Again, I'd suggest adding some error handling here.

File: server/api/events.spec.js [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+    })
+    it('changes name', async () => {
+      return request(app)
+        .put('/api/events/1')
+        .send({name: 'EventUpdated', status: 'pending'})
+        .expect(res => {
+          expect(res.body.name).to.be.equal('EventUpdated')
+          expect(res.body.status).to.be.equal('pending')
+          //description should be set to null
+          expect(res.body.description).to.be.equal(null)
```
Comment (pckilgore) [inline server/api/events.spec.js]: Kinda surprised these pass without returning:true set on the endpoint db queries. Am I missing something?

File: server/api/events.js (hunk 2 above)
Comment (kevslim) [reply server/api/events.js]: We could do the blob thing where we change spaces into underscores

File: server/api/events.js (hunk 8 above)
Comment (kevslim) [reply server/api/events.js]: https://sequelize-guides.netlify.com/inserting-updating-destroying/#modelupdate

**Pull request merged**